src/jdataview.js

```diff
@@ -1,833 +1,471 @@
-		wideValue |= (bitLength < 32 ? (value & ~(-1 << bitLength)) : value) << -this._bitOffset; // setting bits
-
-		for (var i = b.length - 1; i >= 0; i--) {
-			b[i] = wideValue & 0xff;
-			wideValue >>>= 8;
-		}
+	/**
+	 * Sets a 64 bit integer. Takes a regular Number, not a BigInt.
+	 *
+	 * For more precision, use `setBigInt64()`
+	 */
+	setInt64(byteOffset, value, littleEndian) {
+		// Pointer will be handled for us by the bigInt method
+		this.setBigInt64(byteOffset, BigInt(value), littleEndian);
+	}
 
-		this._setBytes(data.start, b, true);
+	/**
+	 * Get an unsigned 64 bit integer. Returns a regular Number, not a BigInt.
+	 *
+	 * For more precision, use `getBigUint64()`
+	 */
+	getUint64(byteOffset, littleEndian) {
+		// Pointer will be handled for us by the bigInt method
+		return Number(this.getBigUint64(byteOffset, littleEndian));
 	}
+
+	/**
+	 * Get a 64 bit integer. Returns a regular Number, not a BigInt.
+	 *
+	 * For more precision, use `getBigInt64()`
+	 */
+	getInt64(byteOffset, littleEndian) {
+		// Pointer will be handled for us by the bigInt method
+		return Number(this.getBigInt64(byteOffset, littleEndian));
+	}
+}
+export default jDataView;
+
+const builtInTypeBytes = {
+	Float64: 8,
+	Float32: 4,
+	BigInt64: 8,
+	BigUint64: 8,
+	Int32: 4,
+	Uint32: 4,
+	Int16: 2,
+	Uint16: 2,
+	Int8: 1,
+	Uint8: 1,
 };
 
-if (NODE) {
-	var nodeNaming = {
-		'Int8': 'Int8',
-		'Int16': 'Int16',
-		'Int32': 'Int32',
-		'Uint8': 'UInt8',
-		'Uint16': 'UInt16',
-		'Uint32': 'UInt32',
-		'Float32': 'Float',
-		'Float64': 'Double'
+const supportedTypes = [
+	...Object.keys(builtInTypeBytes),
+	"Int64",
+	"Uint64",
+	"Signed",
+	"Unsigned",
+	"String",
+	"Char",
+	"Bytes",
+];
+
+// Encapsulate all the built-in methods
+for (const type in builtInTypeBytes) {
+	const typeByteLength = builtInTypeBytes[type];
+	// Getters
+	jDataView.prototype["get" + type] = function (
+		byteOffset = this.tell(),
+		littleEndian = this.littleEndian
+	) {
+		if (typeof this === "undefined") {
+			throw new Error(
+				"'this' is undefined.\nMake sure you call `view.get" +
+					type +
+					"()` directly, without assigning it to a temporary variable, as doing so causes issues."
+			);
+		}
+
+		// Move pointer forwards
+		this.seek(byteOffset + typeByteLength);
+
+		return this.dataView["get" + type](byteOffset, littleEndian);
 	};
 
-	proto._nodeBufferAction = function (type, isReadAction, byteOffset, littleEndian, value) {
-		// Move the internal offset forward
-		this._offset = byteOffset + dataTypes[type];
-		var nodeName = nodeNaming[type] + ((type === 'Int8' || type === 'Uint8') ? '' : littleEndian ? 'LE' : 'BE');
-		byteOffset += this.byteOffset;
-		return isReadAction ? this.buffer['read' + nodeName](byteOffset) : this.buffer['write' + nodeName](value, byteOffset);
-	};
-}
+	// Setters
+	jDataView.prototype["set" + type] = function (
+		byteOffset = this.tell(),
+		value,
+		littleEndian = this.littleEndian
+	) {
+		if (typeof this === "undefined") {
```

Comment:
Hm I don't think this check is worth it here and below. Most JS devs know that you can't just take `obj.method` and call it later without the instance. And if not, the strict mode will catch it pretty quickly for them as `this.seek` below will fail with `this === undefined`.

Reply:
Okay, just something I ran into when testing in a node / deno project.